### PR TITLE
Us 029

### PR DESCRIPTION
**Pull request by weesteeky**
## Description
Please explain briefly the changes you made here.

## Checklist
- [x] All manual security tests passing (Described here: https://github.com/Broom94/Klattice/wiki/Manual-Security-Testing)
- [x] All unit tests passing 
- [x] All UI tests passing
- [x] Code coverage suitable
- [x] WAVE toolbar run on new or updated screens
- [x] Branch named {feature|hotfix|task}/{US-.*}
- [x] If your pull request depends on any other, please link them
- [ ] Approval from Tech lead
- [x] Approval from team member
- [x] Commit messages are meaningful
- [x] Target is green https://travis-ci.com/Broom94/Klattice/jobs/216924893 
- [x] Add `DO NOT MERGE` if you want to postpone merge (in labels)

### CODE REVIEW

File: klattice/src/app/components/login/login.component.html [hunk 1]
```diff
@@ -21,6 +21,7 @@ <h3>Login</h3>
           />
         </div>
         <div class="form-group">
+
```
Comment (jamescrone) [inline klattice/src/app/components/login/login.component.html]: remove whitespace

File: klattice/src/app/components/home/home.component.ts [hunk 2]
```diff
@@ -58,6 +60,7 @@ export class HomeComponent implements OnInit {
 
   data: DataService;
 
+
```
Comment (jamescrone) [inline klattice/src/app/components/home/home.component.ts]: please remove whitespace

File: klattice/src/app/app.component.ts [hunk 3]
```diff
@@ -29,6 +30,7 @@ export class AppComponent implements OnInit {
     this.data.loginDetails.subscribe(x => {
       this.isLoggedIn = x["loggedIn"];
     });
+
```
Comment (jamescrone) [inline klattice/src/app/app.component.ts]: same here

Comment (weesteeky) [reply klattice/src/app/app.component.ts]: got you

File: klattice/src/app/components/home/home.component.ts (hunk 2 above)
Comment (weesteeky) [reply klattice/src/app/components/home/home.component.ts]: fixed

File: klattice/src/app/components/login/login.component.html (hunk 1 above)
Comment (weesteeky) [reply klattice/src/app/components/login/login.component.html]: fixed

File: api/app.js [hunk 4]
```diff
@@ -116,38 +93,44 @@ app.post("/signout", function(req, res) {
...
+        session.isAdmin = true;
+        session.loggedIn = true;
+        session.username = userName;
+        break;
+      case 'e':
+        session.isAdmin = false;
+        session.loggedIn = true;
+        session.username = userName;
+          break;
+    }
```
Comment (matthewdoreilly) [inline api/app.js]: You can remove this Switch statement, and change it to:

```
session.loggedIn = true;
session.username = userName;

if (ans === 'a') {
     session.isAdmin = true;
}
```

Also I'd rename `userName` to `username`, and change your `ans` variable to something more meaningful, like `userRole`

File: api/db.js [hunk 5]
```diff
@@ -7,8 +7,23 @@ const db = mysql.createConnection({
   database: "klattice"
 });
 
-const getUserQuery =
-  "SELECT user_name, user_password, user_type FROM user WHERE user_name = ? AND user_password = ?";
+exports.getUser = function (userName, userPassword){   
+    return new Promise(function(resolve, reject) {
+    var queryValidateUserExists = "SELECT user_name, user_password, user_type FROM user WHERE user_name = ? AND user_password = ?;";
+    db.query(
+        queryValidateUserExists, [userName, userPassword],
```
Comment (matthewdoreilly) [inline api/db.js]: `username` rather than `userName`

File: api/db.js [hunk 6]
```diff
@@ -7,8 +7,23 @@ const db = mysql.createConnection({
...
+    return new Promise(function(resolve, reject) {
+    var queryValidateUserExists = "SELECT user_name, user_password, user_type FROM user WHERE user_name = ? AND user_password = ?;";
+    db.query(
+        queryValidateUserExists, [userName, userPassword], 
+        function (err, rows)
+        {
+            result ='f';
+            if (err) throw err;
+            if(rows.length > 0 && rows[0].user_type == 'admin') result = 'a'; 
+            else if(rows.length > 0 && rows[0].user_type == 'employee') result = 'e';
```
Comment (matthewdoreilly) [inline api/db.js]: Format these if statements like

```
if (...) {
   ...
}
```

Bit more readable

File: klattice/src/assets/saltedHashing.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+exports.saltedHash = function(password)
+{
+    var salt = "S@E1F53135E559C253assdk100101"; 
+    var newPassword = password;
+    newPassword+= salt;
+    newPassword = hash.SHA224(newPassword);
+
+    var passStr = new String(newPassword);
+    return passStr;
```
Comment (matthewdoreilly) [inline klattice/src/assets/saltedHashing.js]: Can just do `return new String(newPassword);` here. Don't need the variable assignment

File: api/app.js (hunk 4 above)
Comment (weesteeky) [reply api/app.js]: got you

File: api/db.js (hunk 5 above)
Comment (weesteeky) [reply api/db.js]: done

File: api/db.js (hunk 6 above)
Comment (weesteeky) [reply api/db.js]: done

File: klattice/src/assets/saltedHashing.js (hunk 7 above)
Comment (weesteeky) [reply klattice/src/assets/saltedHashing.js]: done

**Pull request merged**